doc/news/CMakeLists.txt

```diff
@@ -0,0 +1,55 @@
+## ---------------------------------------------------------------------
+##
+## Copyright (C) 2016 by the deal.II Authors
+##
+## This file is part of the deal.II library.
+##
+## The deal.II library is free software; you can use it, redistribute
+## it, and/or modify it under the terms of the GNU Lesser General
+## Public License as published by the Free Software Foundation; either
+## version 2.1 of the License, or (at your option) any later version.
+## The full text of the license can be found in the file LICENSE at
+## the top level of the deal.II distribution.
+##
+## ---------------------------------------------------------------------
+
+
+#
+# Set up all necessary bits for the changelog generation
+#
+IF(DEAL_II_COMPONENT_DOCUMENTATION)
+
+  MESSAGE(STATUS "")
+  MESSAGE(STATUS "Setting up changelog generation")
+
+  #
+  # Generate changes.h.
+  #
+  ADD_CUSTOM_COMMAND(
+    OUTPUT ${CMAKE_CURRENT_BINARY_DIR}/changes.h
+    COMMAND ${CMAKE_COMMAND} -DOUTPUT_FILE=${CMAKE_CURRENT_BINARY_DIR}/changes.h
+                             -P ${CMAKE_CURRENT_SOURCE_DIR}/changes/create_changes_h.cmake
```

Comment:
Thinking about this again, I don't think this is quite right. You say that the output of this command is `changes.h`, but `changes.h` is really only updated if the fragments actually changed. I think it would be more correct if you said that the output is `changes.h.in`, and then moved the `CONFIGURE_FILE` call from the script you call here to after the current command.

(Makefiles are hard :-( )